src/managers/ApplicationCommandPermissionsManager.js

```diff
@@ -0,0 +1,409 @@
+      roles.forEach(role => {
+        if (typeof role === 'string') {
+          resolvedIDs.push(role);
+          return;
+        }
+        if (!this.guild) throw new Error('GUILD_UNCACHED_ROLE_RESOLVE');
+        const roleID = this.guild.roles.resolveID(role);
+        if (!roleID) throw new TypeError('INVALID_ELEMENT', 'Array', 'users', role);
+        resolvedIDs.push(roleID);
+      });
+    } else if (roles) {
+      if (typeof roles === 'string') {
+        resolvedIDs.push(roles);
+      } else {
+        if (!this.guild) throw new Error('GUILD_UNCACHED_ROLE_RESOLVE');
+        const roleID = this.guild.roles.resolveID(roles);
+        if (!roleID) {
+          throw new TypeError('INVALID_TYPE', 'users', 'Array or RoleResolvable');
+        }
+        resolvedIDs.push(roleID);
+      }
+    }
+    const permissions = existing.filter(perm => !resolvedIDs.includes(perm.id));
+
+    return this.set({ guild: guildID, command: commandID, permissions });
+  }
+
+  /**
+   * Options used to check existance of permissions on a command
+   * <warn>The `command` parameter is not optional when the managers `commandID` is `null`</warn>
+   * @typedef {BaseApplicationCommandPermissionsOptions} HasApplicationCommandPermissionsOptions
+   * @param {UserResolvable|RoleResolvable} permissionID The user or role to check if a permission exists for
+   * on this command.
+   */
+
+  /**
+   * Check whether a permission exists for a user or role
+   * @param {AddApplicationCommandPermissionsOptions} options Options used to check permissions
+   * @returns {Promise<boolean>}
+   * @example
+   * // Check whether a user has permission to use a command
+   * guild.commands.permissions.has({ command: '123456789012345678', permissionID: '876543210123456789' })
+   *  .then(console.log)
+   *  .catch(console.error);
+   */
+  async has({ guild, command, permissionID }) {
+    const { guildID, commandID } = this._validateOptions(guild, command);
+    if (!commandID) throw new TypeError('INVALID_TYPE', 'command', 'ApplicationCommandResolvable');
+
+    let existing = [];
+    try {
+      existing = await this.fetch({ guild: guildID, command: commandID });
+    } catch (error) {
+      if (error.code !== 404) throw error;
+    }
+
+    let resolvedID = permissionID;
+    if (typeof permissionID !== 'string') {
+      resolvedID = this.client.users.resolveID(permissionID);
+      if (!resolvedID) {
+        if (!this.guild) throw new Error('GUILD_UNCACHED_ROLE_RESOLVE');
+        resolvedID = this.guild.roles.resolveID(permissionID);
+      }
+      if (!resolvedID) {
+        throw new TypeError('INVALID_TYPE', 'permissionID', 'UserResolvable or RoleResolvable');
+      }
+    }
+    return existing.some(perm => perm.id === resolvedID);
+  }
+
+  _validateOptions(guild, command) {
+    const guildID = this.guildID ?? this.client.guilds.resolveID(guild);
+    if (!guildID) return Promise.reject(new Error('GLOBAL_COMMAND_PERMISSIONS'));
```

Comment:
You do not await the result of this method, so the rejected promise will just be discarded due to destructuring assignment being used (and probably cause an unhandled promise rejection).

Since this method is only called from within async functions, throwing here should work fine:
```suggestion
    if (!guildID) throw new Error('GLOBAL_COMMAND_PERMISSIONS');
```

Reply:
Yep, discovered this during testing...which is now done